### ISSUE TITLE
Invariant prediction

### ISSUE THREAD

**Issue by zeka0** [fchollet/keras#6847]
I used batch to train a conv neural network, however when I used:
`result = model.predict(x=data, batch_size=batch_size, verbose=True)`
to predict results based on data, the results(has the shape of (18000,  4)) are just 18000 same tuples(shape (4,)).

What's odd is that, data fed to the neural network consists of different images,
and when I inspected the weights of the network, there are no NaNs, meaning it should be impossible for the network to produce same results.

Any idea what could be wrong?
Or how to debug it?

Edit:
I used softmax as the last activation function and binary_crossentropy as the loss function.

**Comment by zeka0** [comment]
I found out what's wrong, not Keras's fault.
